src/main/java/com/google/sps/data/UserAccount.java

```diff
@@ -164,6 +165,12 @@ public static UserAccount fromEntity(Entity entity) {
             : new Mentor(entity);
   }
 
+  protected void validate() {
```

Comment:
I'd expect a function like validate() to perform only non-mutating checks and return a bool. Something like sanitizeData() or checkAndSanitize() might convey the intention better.